### ISSUE TITLE
undefined method 'alias_method_chain' while upgrading to Rails 5.1

### ISSUE THREAD

**Issue by g33kidd** [plataformatec/devise#4509]
I'm in the process of upgrading my Rails app to 5.1.0. Currently having an issue with devise, I think. This is what I get after trying `rails s`. Not really sure what's going on here, but hopefully somebody else here does.

```
~/.rvm/gems/ruby-2.3.3/gems/bundler-1.14.6/lib/bundler/runtime.rb:94:in `rescue in block (2 levels) in require': There was an error while trying to load the gem 'devise'. (Bundler::GemRequireError)
Gem Load Error is: undefined method `alias_method_chain' for ActionDispatch::Routing::RouteSet:Class
Did you mean?  alias_method
```

**Comment by g33kidd** [comment]
Seems I somehow landed at requiring devise 1.5.4 somehow.... hmm.

**Comment by texpert** [comment]
It is restricted to Rails < 5.1 yet:

```ruby
Bundler could not find compatible versions for gem "railties":
  In Gemfile:
    devise (~> 4.2.1) was resolved to 4.2.1, which depends on
      railties (< 5.1, >= 4.1.0)
```

**Comment by g33kidd** [comment]
Yes I was having this issue as well. Still waiting on #4444 i guess?
